### ISSUE TITLE
Support for AIX platform

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-collector-contrib#19195]
### Is your feature request related to a problem? Please describe.

The collector supports amd64, arm64 and ppc64le. I would like to see it support AIX as well.

### Describe the solution you'd like

I would like to see how we can go about supporting AIX. I see the following tasks as part of this work:
* [ ] Identify an environment that can run on AIX
* [ ] Compile and run the equivalent tests we run for ppc64 for AIX
* [ ] Ensure that some receivers such as the hostmetricreceiver work well for AIX
* [ ] Distribute the AIX distribution

**Comment by codeboten** [comment]
Would be good to understand the impact the code of testing the repo on another platform on the duration of the builds. If its negligible, and the existing tools to produce the release supports aix, i don't see any problem with adding support for this.

**Comment by mx-psi** [comment]
Support for new platforms has caused pain in the past and we have been somewhat inconsistent in what we accept (see e.g. open-telemetry/opentelemetry-collector-releases/issues/170, open-telemetry/opentelemetry-collector/issues/2437, open-telemetry/opentelemetry-collector-releases/issues/105 or the issues related to open-telemetry/opentelemetry-collector/issues/5031). We also have different levels and definitions of support for different platforms, that range from "we will accept PRs targeting that platform but won't do anything otherwise" (e.g. Plan 9, see my notes here https://github.com/open-telemetry/opentelemetry-collector/pull/6924#pullrequestreview-1250396643) to "we will provide builds, run unit tests on every PR"(e.g. amd64/linux), passing through "we will make sure it remains buildable, but won't run tests" (e.g. everything macOS). IMO, before discussing AIX specifically we should take a step back and define our support 'tiers'.

I like [`rustc`'s target tier policy](https://doc.rust-lang.org/rustc/target-tier-policy.html), the situation is not identical but probably we can take something from there. The items that we can think in our case about providing support/not on are (roughly from most demanding to least demanding):
- Being able to block a release or make a bugfix release because of problem specific to a platform (i.e. fulfilling the *"The bug is not specific to an uncommon platform"* item from [the bugfix release criteria](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md#bugfix-release-criteria))
- Ensuring feature parity with a reference implementation (amd64/linux) vs. allowing for dummy no-op implementations on components
- Ensuring unit tests pass on CI
- Ability to address platform-specific bugs (e.g. by having an 'owner' of the platform)
- Building and releasing official artifacts for that platform
- Ensuring the codebase builds on a given platform

My take is that we need to write a doc with our tiers (we can keep it simple for now, just try to roughly reflect existing practice), and define a way to move up the tier ladder (at some tier this probably means 'open an issue and get a sponsor that will own things').

**Comment by mx-psi** [comment]
We now have a tiered platform support model, so we can start with adding tier 3 support for AIX now :) :tada:

**Comment by rrschulze** [comment]
We did some initial evaluation to build the core collector on AIX 7.2. There is at least a dependency on [this issue](https://github.com/fsnotify/fsnotify/issues/353) for fsnotify on AIX that needs resolution first.

**Comment by grhoades** [comment]
If we were to supply the infrastructure through a self-service platform that we [operate](https://convergetp.com/converge-ip4g/) would that be helpful?

**Comment by mx-psi** [comment]
@grhoades We now have a [platform support doc](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/platform-support.md). Adding support for AIX would mean starting from Tier 3 (guaranteed to build) and working from there. You can take a look at the PRs related to `linux/s390x` on this, the opentelemetry-collector and the opentelemetry-collector-releases repositories to see an example of the work needed for this.

**Comment by ThakkarChandresh** [comment]
Raised PR with working code of it!

**Comment by FreerideTheWorld** [comment]
This thread seems to be a bit out of my league but if there was a way to get OpenTelemtry collector on AIX and ship logs to Elasticsearch that would be very helpful.  Sorry if this isn't a great place to post.  I do understand the hurdles of development here since the AIX platform isn't something that most people have access to for development/testing.

**Comment by mx-psi** [comment]
As a summary: [Tier 3 support](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/platform-support.md#tier-3---community-support) for AIX can be added if someone leads the development and owns it. This is probably still something we need to address in some way (either with reduced functionality or by adding support for this on the dependency) https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19195#issuecomment-1779075809.

Tier 3 support only guarantees that the Collector builds on AIX, but the Collector SIG will generally not provide direct fixes to AIX-specific issues after this is added.

**Comment by Vitor0Mateus00** [comment]
Hi everyone! My name is Vitor, I'm an Platform Engineer working with observability in 
enterprise environments with daily access to production AIX machines (financial/telecom sector).

I recently opened #43106 as a proposal, which was redirected here. I'd like to be the 
**owner and driver** of Tier 3 AIX support, which from what I understand is the main blocker 
for moving forward.

A few things I can bring to the table:
- Access to production AIX machines for building and testing
- Willingness to own AIX-specific issues going forward

Before I start, I want to make sure I'm not duplicating existing work:
- **@ThakkarChandresh** — your PR from October 2024 was mentioned but I couldn't find it. Is it still active? Would you be open to collaborating?
- **@grhoades** — is your offer of self-hosted AIX CI infrastructure from August 2024 still available? That would be critical for automating builds.
- Is the `fsnotify` dependency still the main technical blocker, or have there been other findings since then?

I'm ready to start from the s390x work as a reference and open a PR targeting Tier 3 
(build guarantee only). Just need a sponsor/maintainer to guide the review process.

**Comment by mx-psi** [comment]
I am happy to be pinged on PRs related to this for reviews

**Comment by Dylan-M** [comment]
I've also been working on supporting AIX, and still interested in being one of the AIX maintainers. See my post here:
https://github.com/open-telemetry/opentelemetry-collector/issues/13380#issuecomment-4006449596

I've already made significant progress, and it will be continued going forward.

**Comment by sswamy48** [comment]
Dear Team,

We can arrange access to an AIX Lpar in the OSU Lab. We can expect the setup to be available in 2-3 week.
In parallel, we have been testing the hostmetrics receiver by using the changes from @Dylan-M .
We were able to get most of the metrics and validating the metrics returned.

**Comment by Dylan-M** [comment]
@atoulme And I are working on the Tier 3 support change now. If you're in favor, and can approve, please do so (or request changes).

If you cannot do a review, you can give it a 👍 to show your support.

https://github.com/open-telemetry/opentelemetry-collector/pull/14793

I'll be tackling CI in contrib next week, so that we can move contrib to Tier 3 as well.

**Comment by Dylan-M** [comment]
> Dear Team,
> 
> We can arrange access to an AIX Lpar in the OSU Lab. We can expect the setup to be available in 2-3 week. In parallel, we have been testing the hostmetrics receiver by using the changes from [@Dylan-M](https://github.com/Dylan-M) . We were able to get most of the metrics and validating the metrics returned.

If you want to chat, you can find me on the CNCF Slack under `Dylan Myers`.

**Comment by douglascamata** [comment]
Hey folks! I'm reviewing @atoulme's work on https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1424 to start releasing artifacts for  `aix/ppc64`. It seems like we have an issue with `github.com/cockroachdb/pebble/v2/vfs` on this platform:

> Error: │ /home/runner/go/pkg/mod/github.com/cockroachdb/pebble/v2@v2.1.6/vfs/vfs.go:235:18: cannot use defaultFS{} (value of struct type defaultFS) as FS value in variable declaration: defaultFS does not implement FS (missing method GetDiskUsage)

Build with more details: https://github.com/open-telemetry/opentelemetry-collector-releases/actions/runs/27626597035/job/81699448187?pr=1424.

I think this is related to the recent work on https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/tailstorage/pebbletailstorageextension. 

We need a solution for this.

**Comment by douglascamata** [comment]
To provide more information, @songy23 tried something at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/48967 and it seems to be successful at fixing the binary's compilation. But when adding that extension to a distribution and building it for `aix`, there are compilation errors.

**Comment by Dylan-M** [comment]
We (Bindplane) had to fix this by having a separate manifest for AIX that excludes components that don't build properly for it. As far as I can tell, the current build system using a manifest has no way to specify something is for a specific OS only.

**Comment by Dylan-M** [comment]
It looks like this should be fairly easy to support on the Pebble side, so it wouldn't need to be a permanent exclusion (nor would we want it to be).

**Comment by Dylan-M** [comment]
@douglascamata I've filed an [issue](https://github.com/cockroachdb/pebble/issues/6099) with Pebble, along with a [PR fixing it](https://github.com/cockroachdb/pebble/pull/6100). Hopefully they accept it. If they do, we could use a replace statement to pull it in until a new Pebble release drops including it.

**Comment by Dylan-M** [comment]
Looks like they will accept it, one approval already in the books.

**Comment by douglascamata** [comment]
Lovely. Thank you, @Dylan-M!

**Comment by Dylan-M** [comment]
> Lovely. Thank you, [@Dylan-M](https://github.com/Dylan-M)!

And it is now merged. Talk about timing.

Let me know if there is anything else I can do to help.

**Comment by douglascamata** [comment]
@Dylan-M: I think all we will need is to bump the version of pebble we use in the extension here. This should unblock the required changes in the releases repo.

**Comment by Dylan-M** [comment]
@douglascamata Since they haven't done a release, it would need to be a direct to GitHub SHA replace statement I think.

Edit: Or a direct to SHA `go get` pin.

**Comment by douglascamata** [comment]
If the maintainers of the component are okay with pinning to a commit SHA, we can go with that. Otherwise we have two options:

1. Proceed with the AIX release for all distros except contrib and add it later. 
2. AIX stays blocked.

I'm not involved in the AIX support to know whether have a build for contrib is what really matters (I'm just helping to get it done), but I assume starting with the other distros can be useful already.

I leave the decision up to the people interested. 😄

**Comment by perk** [comment]
Interestingly, IBM released the otelcol-contrib in their `AIX Toolbox` quite recently (June 18th) as well: https://community.ibm.com/community/user/discussion/opentelemetry-cpp-and-collector-is-now-available-on-aix-toolbox

* https://www.ibm.com/support/pages/aix-toolbox-open-source-software-downloads-alpha#O
* https://www.ibm.com/support/pages/node/6833478

**Comment by atoulme** [comment]
The work took place [here](https://github.com/open-telemetry/opentelemetry-collector/issues/13380) and we now have a complete distribution with contrib for AIX. Closing as complete.
